### DISCUSSION TITLE
Cross-compiling for x86 Synology from Apple Silicon Mac

### DISCUSSION THREAD

**Discussion by rgov** [runfalk/synology-wireguard#140] [General]
Docker Desktop for Apple Silicon supports running x86 containers under emulation. This requires building the container with `docker buildx build` and then specifying the `--platform` option to `docker run`. It will of course take longer than on the native architecture.

    $ docker buildx build --platform linux/amd64 --output type=image,push=false --tag runfalk/synobuild .

    $ docker run --rm --platform linux/amd64 --entrypoint /bin/uname runfalk/synobuild -a
    Linux 01a5fb73630e 5.10.76-linuxkit #1 SMP PREEMPT Mon Nov 8 11:22:26 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

    $ docker run --rm -it --platform linux/amd64 --privileged \
        --env PACKAGE_ARCH=geminilake \
        --env DSM_VER=7.0 \
        -v $(pwd)/toolkit_tarballs:/toolkit_tarballs \
        -v $(pwd)/artifacts:/result_spk \
        runfalk/synobuild

**Comment by fabiov64** [comment]
On Silicon Mac it's possible to cross compile for all the platforms, just following the standard approach. As reported in my post: https://github.com/runfalk/synology-wireguard/discussions/138 I successfully compiled for armada38x. I assume it's possible to compile for all the other platforms. The important thing is to pull all the required patches for a given platform, that are not yet included in the master.
